js/utils/tabFunctions.js

```diff
@@ -37,8 +37,42 @@ function getTabDivPair (tabN, tablayoutId) {
     let div_tab = document.createElement("div");
     div_tab.id = `layout${tablayoutId}.file${tabN}`;
     div_tab.className = `tabcontent belongs_to_layout_${tablayoutId}`;
-    div_tab.contentEditable="true";
-    div_tab.innerHTML = `Start typings in file - ${tabN}`;
+    // div_tab.classList.add("editor");
+    // div_tab.contentEditable="true";
+
+    
+    // Create a div for linenumbers with the class "line-numbers"
+    var lineNumbersDiv = document.createElement("div");
+    lineNumbersDiv.classList.add("line-numbers");
+    lineNumbersDiv.innerHTML = '<span></span>';
+    
+    // Create a textarea element
+    var textarea = document.createElement("textarea");
+    textarea.innerHTML = `Start typings in file - ${tabN}`;
+
+    // Append the lineNumbersDiv and textarea to the parentDiv
+    div_tab.appendChild(lineNumbersDiv);
+    div_tab.appendChild(textarea);
+
+    textarea.addEventListener('keyup', event => {
+        const numberOfLines = event.target.value.split('\n').length

```

Comment:
@code-master-ajay Such code might work for small files but be a hassle during load stage of large files. Also, this will keep getting laggier as the size of the file grows. See if you can think of a better way of doing this.

Reply:
Also, I believe this will get called on every keyup event and we don't have a filter of the type of key press only during which this should happen. This will do uneccesary span re-filling and re-rendering on every key up. @code-master-ajay

Reply:
@bot2x will try to find other, more efficient approach.